libs/cpp-ledger/ledger/ledger.hpp

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <memory>
+#include <tuple>
+#include "cpp-ledger/common/types.hpp"
+
+namespace ledger {
+
+  class LedgerDevice {
+   public:
+    virtual ~LedgerDevice() = default;
+
+    virtual std::tuple<Bytes, Error> Exchange(const Bytes &command) const = 0;
+    virtual void Close() = 0;
+  };
+
+  class LedgerAdmin {
+   public:
+    virtual ~LedgerAdmin() = default;
+
+    virtual int CountDevices() const = 0;
+    virtual std::tuple<std::string, Error> ListDevices() const = 0;
```

Comment:
what does this function do?  
what should it return?  
should it write to stdout?

Reply:
This function prints all HID devices to defined out stream. By default, the lib user could print it to terminal.
It should return empty string for HID devices.

There is also Zemu (Zondax Zemu Testing Framework) implementation of Ledger device works via RPC.
In this case ListDevice() method should return "Zemu device" string and nothing to terminal.

We don't need Zemu implementation yet, but I decided to keep original interface of this method.